x509/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+/*++
+Licensed under the Apache-2.0 license.
+
+Abstract:
+    DPE x509 Library Crate.
+--*/
+#![cfg_attr(not(test), no_std)]
+
+pub mod x509;
+pub mod tci;
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum X509ErrorCode {
+    NoError = 0,
+    InternalError = 1,
+    InvalidCommand = 2,
+    InvalidArgument = 3,
+}
+
+pub enum DpeProfile {
+    P256Sha256 = 1,
+    P384Sha384 = 2,
+}
+
+impl DpeProfile {
```

Comment:
I wonder if we should have a `common` crate for stuff like this and errors. Since this isn't really X.509-specific